pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/BaseClusterIntegrationTest.java

```diff
@@ -16,1402 +16,322 @@
+        try {
+          ClusterIntegrationTestUtils.setUpH2TableWithAvro(avroFiles, getTableName(), _h2Connection);
+        } catch (Exception e) {
+          // Ignored
+        }
       }
     });
   }
 
-  public void pushAvroIntoKafka(final List<File> avroFiles, ExecutorService executor, final String kafkaTopic, final byte[] header) {
+  /**
+   * Set up query generator using the given Avro files.
+   *
+   * @param avroFiles List of Avro files
+   * @param executor Executor
+   */
+  protected void setupQueryGenerator(@Nonnull final List<File> avroFiles, @Nonnull ExecutorService executor) {
+    Assert.assertNull(_queryGenerator);
+    final String tableName = getTableName();
     executor.execute(new Runnable() {
       @Override
       public void run() {
-        pushAvroIntoKafka(avroFiles, KafkaStarterUtils.DEFAULT_KAFKA_BROKER, kafkaTopic, header);
+        _queryGenerator = new QueryGenerator(avroFiles, tableName, tableName);
       }
     });
   }
 
-  public void pushAvroIntoKafka(final List<File> avroFiles, ExecutorService executor, final String kafkaTopic) {
-    pushAvroIntoKafka(avroFiles, executor, kafkaTopic, null);
-  }
-
-  public File getSchemaFile() {
-    return new File(OfflineClusterIntegrationTest.class.getClassLoader()
-        .getResource("On_Time_On_Time_Performance_2014_100k_subset_nonulls.schema").getFile());
-  }
-
-
-  protected String getSingleStringValueFromJSONAggregationResults(JSONObject jsonObject) throws JSONException {
-    return jsonObject.getJSONArray("aggregationResults").getJSONObject(0).getString("value");
-  }
-
-  protected JSONArray getGroupByArrayFromJSONAggregationResults(JSONObject jsonObject) throws JSONException {
-    return jsonObject.getJSONArray("aggregationResults").getJSONObject(0).getJSONArray("groupByResult");
-  }
-
-  protected void ensurePinotConnectionIsCreated() {
-    if (_pinotConnection == null) {
-      synchronized (BaseClusterIntegrationTest.class) {
-        if (_pinotConnection == null) {
-          _pinotConnection = ConnectionFactory.fromZookeeper(ZkStarter.DEFAULT_ZK_STR + "/" + getHelixClusterName());
-        }
-      }
-    }
-  }
-
-  protected long getCurrentServingNumDocs(String tableName) {
-    ensurePinotConnectionIsCreated();
-    try {
-      com.linkedin.pinot.client.ResultSetGroup resultSetGroup =
-          _pinotConnection.execute("SELECT COUNT(*) from " + tableName + " LIMIT 0");
-      if (resultSetGroup.getResultSetCount() > 0) {
-        return resultSetGroup.getResultSet(0).getInt(0);
-      }
-      return 0;
-    } catch (Exception e) {
-      return -1L;
-    }
+  /**
+   * Get the schema file.
+   *
+   * @return Schema file
+   */
+  @Nonnull
+  protected File getSchemaFile() {
+    URL resourceUrl = BaseClusterIntegrationTest.class.getClassLoader().getResource(getSchemaFileName());
+    Assert.assertNotNull(resourceUrl);
+    return new File(resourceUrl.getFile());
```

Comment:
Does this actually work properly? I'm pretty sure that when the tests are JAR'ed, this call will fail due to using a JAR URL from which a File cannot be created.